lib/publiccloud/azure_client.pm

```diff
@@ -47,6 +47,7 @@ sub az_login {
     #Azure infra need some time to propagate given by Vault credentials
     # Running some verification command does not prove anything because
     # at the beginning failures can happening sporadically
+    record_info("WAIT", "Waiting for Azure credential spreading") if get_var('AZURE_LOGIN_WAIT_SECONDS', 0);
```

Comment:
This will record info always, since you are adding a default value. 
 ```suggestion
 if (my $wait_seconds = get_var('AZURE_LOGIN_WAIT_SECONDS')) {
     record_info("WAIT", "Waiting for Azure credential spreading");
     sleep($wait_seconds);
 }
    record_info("WAIT", "Waiting for Azure credential spreading") if get_var('AZURE_LOGIN_WAIT_SECONDS', 0);
```